### PR TITLE
Remove Html::getAttributeName + final class + private constants

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    |  ❌
| New feature?  | ❌
| Breaks BC?    | ✔️
| Fixed issues  | -

- Method `Html::getAttributeName()` moved to `yiisoft/form` (https://github.com/yiisoft/form/pull/48)

- Set `Html` as final class

- Set all constants as private

- Add one more test for `Html::escapeJsRegularExpression()`

With this PR tested:

- yiisoft/form (need https://github.com/yiisoft/form/pull/48)
- yiisoft/yii-bootstrap4
- yiisoft/yii-bootstrap5
- yiisoft/yii-bulma

MERGE AFTER https://github.com/yiisoft/form/pull/48

### CODE REVIEW

File: src/Html.php [hunk 1]
```diff
@@ -1635,20 +1650,20 @@ public static function cssStyleToArray(string $style): array
...
-     * If `$attribute` has neither prefix nor suffix, it will be returned back without change.
      * @param string $attribute the attribute name or expression
-     *
-     * @return string the attribute name without prefix and suffix.
-     *
+     * @return array
      * @throws InvalidArgumentException if the attribute name contains non-word characters.
      */
-    public static function getAttributeName(string $attribute): string
+    public static function parseAttribute(string $attribute)
```
Comment (samdark) [inline src/Html.php:1657]: This method looks like internal thing. Does it have to be `public`?

Comment (vjik) [reply src/Html.php:1657]: i'm use this here: https://github.com/yiisoft/form/pull/48/files#diff-367552c756b09b53720eccec0ddf71a4R79

Comment (vjik) [reply src/Html.php:1657]: Hm... maybe make method `Html::wrapAttributeName(string $wrapName, string $attributeName)` and make `parseAttribute` is private:

```php
`Html::wrapAttributeName('myform', 'name[0]'); // myform[name][0]
```

?

Comment (samdark) [reply src/Html.php:1657]: I mean it's not used externally (except test) so likely it is a good idea to make it `private`.

Comment (vjik) [reply src/Html.php]: `parseAttribute` use in yiisoft/form in new PR: https://github.com/yiisoft/form/pull/48/files#diff-367552c756b09b53720eccec0ddf71a4R79

But now I make `parseAttribute` is private and create new method `wrapAttributeName`.

File: src/Html.php [hunk 2]
```diff
@@ -1625,6 +1625,43 @@ public static function cssStyleToArray(string $style): array
...
+    }
+
+    /**
+     * @param string $wrapName
+     * @param string $attributeName
+     * @return string
+     * @throws InvalidArgumentException if the attribute name contains non-word characters.
+     * @throws EmptyWrapNameException if wrap name is empty and attribute name is tabular.
+     */
+    public static function wrapAttributeName(string $wrapName, string $attributeName): string
```
Comment (samdark) [inline src/Html.php:1646]: It's not clear what the method does. What is "wrap"? Also, based on the code, it seems it's not wrapping name but prefixing it.

File: src/Html.php [hunk 3]
```diff
@@ -1625,6 +1625,43 @@ public static function cssStyleToArray(string $style): array
 
     /**
      * Returns the real attribute name from the given attribute expression.
+     * If `$attribute` has neither prefix nor suffix, it will be returned back without change.
+     * @param string $attribute the attribute name or expression
+     * @return string the attribute name without prefix and suffix.
+     * @throws InvalidArgumentException if the attribute name contains non-word characters.
+     * @see Html::parseAttribute()
+     */
+    public static function getAttributeName(string $attribute): string
```
Comment (samdark) [inline src/Html.php:1634]: Do you mean removing index here?

File: tests/HtmlTest.php [hunk 4]
```diff
@@ -1229,6 +1230,44 @@ public function testGetAttributeName(): void
         Html::getAttributeName('content body');
     }
 
+    public function wrapAttributeNameData(): array
+    {
+        return [
+            ['form', '[0]content', 'form[0][content]'],
```
Comment (samdark) [inline tests/HtmlTest.php:1236]: What does `[0]content` mean in an attribute name?

File: src/Html.php (hunk 2 above)
Comment (vjik) [reply src/Html.php:1646]: I try move [this code](https://github.com/yiisoft/form/blob/67f40a2ddb6bdbb75e86ea431dff9596a239dab9/src/Helper/HtmlForm.php#L87-L101) to Html Helper... Has idea how do name the method?

Comment (samdark) [reply src/Html.php:1646]: Hmm... it looks very form-packge-specific. Is there a use case for it outside of form package?

File: src/Html.php [hunk 5]
```diff
@@ -22,18 +22,18 @@
  * You can specify, for example, `class`, `style` or `id` for an HTML element using the `$options` parameter. See the
  * documentation of the {@see tag()} method for more details.
  */
-class Html
+final class Html
```
[the remark this answers is not in the document]
Comment (samdark) [reply src/Html.php:25]: Why not? There's no sense in extending it.

File: src/Html.php (hunk 2 above)
Comment (vjik) [reply src/Html.php:1646]: > Hmm... it looks very form-packge-specific. Is there a use case for it outside of form package?

We have method `getAttributeName` which also work with attribute expressions such as `[0]content` (used in tabular data input to represent the "content" attribute for the first model in tabular input), `dates[0]` (represents the first array element of the "dates" attribute), `[0]dates[0]` (represents the first array element of the "dates" attribute for the first model in tabular input).

Why not have a method (`wrapAttributeName`) for this functionality?

Comment (vjik) [reply src/Html.php:1646]: Hm.. It looks like `getAttributeName` also is only needed in `yiisoft/form`.

File: tests/HtmlTest.php (hunk 4 above)
Comment (vjik) [reply tests/HtmlTest.php]: Update name and description of this PR.

**Review by samdark — APPROVED** [review]

**Review by samdark — APPROVED** [review]

**Pull request merged**